docker/adamant_env.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+this_dir=`readlink -f "${BASH_SOURCE[0]}" | xargs dirname`
+
+set +e
+
+if ! command -v docker &> /dev/null
+then
+  if command -v podman &> /dev/null
+  then
+    function docker() {
+      podman $@
+    }
+  else
+    echo "Neither docker nor podman found!!!"
+    exit 1
+  fi
+fi
+
+DOCKER_CONTAINER_NAME="adamant_container"
+DOCKER_IMAGE_NAME="ghcr.io/lasp/adamant:latest"
+DOCKER_COMPOSE_COMMAND="docker compose"
+DOCKER_COMPOSE_CONFIG="${this_dir}/compose.yml"
+export DOCKER_CONTAINER_NAME
+export DOCKER_IMAGE_NAME
+export DOCKER_COMPOSE_COMMAND
+${DOCKER_COMPOSE_COMMAND} version &> /dev/null
+if [ "$?" -ne 0 ]; then
+  export DOCKER_COMPOSE_COMMAND="docker-compose"
+fi
+
+# Helper function to print out command as executed:
+execute () {
+  echo "$ $@"
+  eval "$@"
+}
+
+set -e
+
+usage() {
+  echo "Usage: $1 [start, stop, login, push, build, remove]" >&2
+  echo "*  start: create and start the adamant container" >&2
+  echo "*  stop: stop the running adamant container" >&2
+  echo "*  login: login to the adamant container" >&2
+  echo "*  push: push the image to the Docker registry" >&2
+  echo "*  build: build the image from the Dockerfile" >&2
+  echo "*  remove: remove network and volumes for adamant" >&2
+  exit 1
+}
+
+if [ "$#" -eq 0 ]; then
+  usage $0
+fi
+
+case $1 in
+  start )
+    execute "${DOCKER_COMPOSE_COMMAND} -f ${DOCKER_COMPOSE_CONFIG} up -d"
+    echo ""
+    echo "Run \"./adamant_env.sh login\" to log in."
+    ;;
+  stop )
+    execute "${DOCKER_COMPOSE_COMMAND} -f ${DOCKER_COMPOSE_CONFIG} stop"
+    ;;
+  login )
+    execute "docker exec -it -u user ${DOCKER_CONTAINER_NAME} //bin//bash"
+    ;;
+  push )
+    execute "docker push ${DOCKER_IMAGE_NAME}"
+    ;;
+  build )
+    execute "docker build --progress=plain -t ${DOCKER_IMAGE_NAME} -f Dockerfile ."
```

Comment:
shouldn't this just use `docker compose build`?

Reply:
Addressed in [cdaf5fe](https://github.com/lasp/adamant/pull/50/commits/cdaf5fed1b1dc48981e008a9a621e2d8810fea5c).